react/Savings.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useContext } from 'react'
+import { defineMessages, FormattedNumber } from 'react-intl'
+import { ProductContext } from 'vtex.product-context'
+import { FormattedCurrency } from 'vtex.format-currency'
+import { useCssHandles } from 'vtex.css-handles'
+import { IOMessageWithMarkers } from 'vtex.native-types'
+
+import { StorefrontFC } from './types'
+
+interface Props {
+  message: string
+  markers: string[]
+}
+
+const CSS_HANDLES = [
+  'savings',
+  'previousPriceValue',
+  'newPriceValue',
+  'savingsValue',
+  'savingsPercentage',
+] as const
+
+const Savings: StorefrontFC<Props> = props => {
+  const { message, markers } = props
+  const handles = useCssHandles(CSS_HANDLES)
+  const { selectedItem } = useContext(ProductContext)
+
+  if (!selectedItem) {
+    return null
+  }
+
+  const previousPriceValue = selectedItem.sellers[0].commertialOffer.ListPrice
+  const newPriceValue = selectedItem.sellers[0].commertialOffer.Price
```

Comment:
Is there any chance that `sellers` or `commertialOffer` are `null` or `undefined`?
It doesn't seen likely, but just to be safe :)